packages/js/uri-resolvers/src/utils/buildUriResolver.ts

```diff
@@ -0,0 +1,44 @@
+import { UriResolverAggregator } from "../aggregator";
+import { UriResolverLike } from "../helpers";
+import { PackageResolver } from "../packages";
+
+import { Result } from "@polywrap/result";
+import {
+  IUriResolver,
+  Uri,
+  Client,
+  IWrapPackage,
+  IUriPackage,
+} from "@polywrap/core-js";
+
+export const buildUriResolver = <TError = undefined>(
```

Comment:
real nit please feel free to ignore: since all the conditionals are based on `resolvable` variable, it might make sense to change this to a switch statement to make the code easier to read

Reply:
I don't think this could be rewritten as a switch statement, but if you have an idea, please share!

Reply:
yah no this can't be written as a switch :P